### PR TITLE
Group support

### PR DESCRIPTION
**Pull request by PragTob**
adds the most basic support for groups, along with some fixes. I'm opening another PR on simplecov in a second to also integration test this, as that was quite tough around here.

### CODE REVIEW

**Comment by PragTob** [comment]
@fede-moya any tricks to triggering the CI, it seems it doesn't like me and there is _some_ chance I broke the docker tests but I don't think so.

**Comment by fede-moya** [comment]
Hi @PragTob I will be reviewing this one later today. About,

> any tricks to triggering the CI, it seems it doesn't like me and there is some chance I broke the docker tests but I don't think so.

I guess CircleCi isn't configured to analysis contributions I will try to fix that.

File: spec/fixtures/sample.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "meta": {
-    "simplecov_version": "0.18.5"
+    "simplecov_version": "0.21.2"
   },
   "coverage": {
-    "/gem/spec/fixtures/sample.rb": {
+    "/stub_working_directory/spec/fixtures/sample.rb": {
```
Comment (fede-moya) [inline spec/fixtures/sample.json:6]: 👍🏼

File: spec/simplecov_json_formatter_spec.rb [hunk 2]
```diff
@@ -14,14 +14,14 @@
   end
 
   describe 'format' do
-    context 'whit line coverage' do
+    context 'with line coverage' do
```
Comment (fede-moya) [inline spec/simplecov_json_formatter_spec.rb:17]: thanks

**Review by fede-moya — COMMENTED** [review]
@PragTob  Thanks for the house cleaning 🧹 . About your comment around integrations tests, looks like you have a point. Do you think worths adding some integration tests here right ? I could work on that to make this library more reliable. About the CI, CircleCi is now configure to analysis PRs from forked repos so if you trigger the CI with a new commit it should run. Holding approval till we are sure tests pass.

**Comment by PragTob** [comment]
@fede-moya it's still my plan to take this and move it entirely to a simplecov mono repo :) Imo it makes it much easier for everyone if simplecov(core), simplecov-html and simplecov-json (yes I'd probably rename and re release it for consistency) lived in the same repo. Like, top level goes into the three projects, just some top level shenanigans to run all tests etc. - so I don't think it needs to be handled here unless you wanna take a stab at that unification over in the simplecov repo :)

**Comment by PragTob** [comment]
for reference, added them over here: https://github.com/simplecov-ruby/simplecov/pull/1015

**Comment by fede-moya** [comment]
@PragTob 
> unless you wanna take a stab at that unification over in the simplecov repo :)

No, I don't. I would take it as a compliment 😃 . Go for it, ping me if you think I can assist in any way, I would be happy to help.

**Comment by PragTob** [comment]
@fede-moya any updates on the CI? I'd love to get this merged and released without completely moving the project :grin:

**Comment by fede-moya** [comment]
@PragTob yeap, as I pointed on https://github.com/codeclimate-community/simplecov_json_formatter/pull/2#pullrequestreview-872421002

>  CircleCi is now configure to analysis PRs from forked repos so if you trigger the CI with a new commit it should run

I believe that's accurate. Can you try that ?

**Comment by PragTob** [comment]
@fede-moya ah, scuzi somehow I missed that :facepalm: Pusching new commit now

**Comment by PragTob** [comment]
yup it's running, beautiful - thank you!

**Comment by fede-moya** [comment]
@PragTob Nice 👍🏼 . I saw it's failing due to rubocop complaining, feel free to make changes to rubocop's rules.

**Review by fede-moya — APPROVED** [review]

**Pull request merged**